internal/config/executionspace.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Axis Communications AB.
+//
+// For a full list of individual contributors, please see the commit history.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"flag"
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type ExecutionSpaceConfig interface {
+	Config
+	Hostname() string
+	Timeout() time.Duration
+	ExecutionSpaceWaitTimeout() time.Duration
+	RabbitMQHookURL() string
+	RabbitMQHookExchangeName() string
+	EiffelGoerURL() string
+}
+
+// executionSpaceCfg implements the ExecutionSpaceConfig interface.
+type executionSpaceCfg struct {
+	Config
+	stripPrefix               string
+	hostname                  string
+	timeout                   time.Duration
+	executionSpaceWaitTimeout time.Duration
+	rabbitmqHookURL           string
+	rabbitmqHookExchange      string
+	eiffelGoerURL             string
+}
+
+// NewExecutionSpaceConfig creates an executio nspace config interface based on input parameters or environment variables.
+func NewExecutionSpaceConfig() ExecutionSpaceConfig {
+	var conf executionSpaceCfg
+
+	defaultTimeout, err := time.ParseDuration(EnvOrDefault("REQUEST_TIMEOUT", "1m"))
+	if err != nil {
+		logrus.Panic(err)
+	}
+
+	executionSpaceWaitTimeout, err := time.ParseDuration(EnvOrDefault("EXECUTION_SPACE_WAIT_TIMEOUT", "1h"))
+	if err != nil {
+		logrus.Panic(err)
+	}
+
+	flag.StringVar(&conf.hostname, "hostname", EnvOrDefault("PROVIDER_HOSTNAME", "http://localhost"), "Host to supply to ESR for starting executors")
+	flag.DurationVar(&conf.timeout, "timeout", defaultTimeout, "Maximum timeout for requests to Execution space provider Service.")
+	flag.DurationVar(&conf.executionSpaceWaitTimeout, "executionSpaceWaitTimeout", executionSpaceWaitTimeout, "Timeout duration to wait when trying to checkout execution space(s)")
+	flag.StringVar(&conf.rabbitmqHookURL, "rabbitmq_hook_url", os.Getenv("ETOS_RABBITMQ_URL"), "URL to the ETOS rabbitmq for logs")
+	flag.StringVar(&conf.rabbitmqHookExchange, "rabbitmq_hook_exchange", os.Getenv("ETOS_RABBITMQ_EXCHANGE"), "Exchange to use for the ETOS rabbitmq for logs")
+	flag.StringVar(&conf.eiffelGoerURL, "event_repository_host", os.Getenv("EIFFEL_GOER_URL"), "Event repository URL used for Eiffel event lookup")
```

Comment:
If the is specifically a GoER url, then the help text etc should not talk about a generic event repository url.

Reply:
While I agree with the comment, this is out of scope for this PR. I only moved these variables from https://github.com/eiffel-community/etos-api/pull/89/commits/9e806d46845ae07d839a69649e67f0cf18b4f4aa#diff-7bcfcaac6de878edc282f448cc09fc79d74566aab79e4fa8a73eb0235b57d0f4 to here.